cmd/zfs/zfs_main.c

```diff
@@ -7755,6 +7755,12 @@ unshare_unmount_path(int op, char *path, int flags, boolean_t is_manual)
 		return (1);
 	}
 
+	if (strcmp(entry.mnt_mountp, path) != 0) {
+		(void) fprintf(stderr, gettext("cannot %s '%s': not an original"
+		    " mountpoint\n"), cmdname, path);
```

Comment:
While I understand there is a problem, I am not sure this solution is right.  Please correct me if I am wrong, but looking on getextmntent() I see this comment:
```
 * Search for the path in /proc/self/mounts. Rather than looking for the
 * specific path, which can be fooled by non-standard paths (i.e. ".."
 * or "//"), we stat() the path and search for the corresponding
 * (major,minor) device pair.
```
, which makes me think the exactly path match is not required there.